packages/@aws-cdk/aws-apigateway/lib/rate-limited-api-key.ts

```diff
@@ -0,0 +1,54 @@
+import { Construct, Resource } from '@aws-cdk/core';
+import { ApiKey, ApiKeyProps, IApiKey } from './api-key';
+import { QuotaSettings, ThrottleSettings, UsagePlan, UsagePlanPerApiStage } from './usage-plan';
+
+/**
+ * RateLimitedApiKey properties.
+ */
+export interface RateLimitedApiKeyProps extends ApiKeyProps {
```

Comment:
Missed something from last time. We usually don't extend one Props class from another Props class.

The following changes are needed here
- Rename `ApiKeyProps` to `ApiKeyOptions`
- Create a new empty `ApiKeyProps` class - 
  ```ts
  export interface ApiKeyProps extends ApiKeyOptions {
  }
  ```

Reply:
Not sure if I understand.

Is the suggestion to :
1. Rename ```ApiKeyProps``` defined in ```ApiKey```, to ```ApiKeyOptions```
2. Create an empty interface (```ApiKeyProps```) in ```RateLimitedApiKey``` extending ```ApiKeyOptions```
3. Use it like 
```typescript
export interface RateLimitedApiKeyProps extends ApiKeyProps {
```

If that's the case, I'm confused how it will help/is better

Reply:
You're correct. I think I was looking for an optimization/future-proofing whose value is unclear yet. Ignore this.